src/components/signup/SignupState.js

```diff
@@ -94,28 +94,54 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
     }
   }
 
+  const getW3UserFromProps = async () => {
+    const web3Token = await AsyncStorage.getItem('GD_web3Token')
+
+    // Getting the second element from routes array (starts from 0) as the second route is Phone
+    // We are redirecting directly to Phone from Auth component if w3Token provided
+    const _w3User = _get(navigation, 'state.routes[1].params.w3User')
+    let w3User = _w3User && typeof _w3User === 'object' ? _w3User : {}
+
+    if (web3Token && Object.keys(w3User).length) {
+      const userScreenData = {
+        email: w3User.email,
+        fullName: w3User.full_name,
```

Comment:
w3User.full_name can be undefined. I think it will be better `w3User.full_name || ""`

Reply:
done